### PR TITLE
Refactor: Consolidate code for query generator

### PR DESCRIPTION
**Pull request by yangkb09**
Paired with @darrenjaneczek to consolidate and refactor query generator related code into `public/app/features/trails/AutomaticMetricQueries/query-generators/default.ts` for readability

- Refactor `generator` / `getGeneratorParameters` into `createDefaultMetricQueryDefs`
- Move general base query related code from `common/queries.ts` to `common/baseQuery.ts`
- Move from `common/queries.ts` to `default.ts`
  - `generateQueries`
  - `aggLabel` related code
- Move logic from `common/rules.ts`, `common/types.ts`, and `types.ts` to `default.ts`

Ref https://github.com/grafana/grafana/issues/90294

### CODE REVIEW

File: public/app/features/trails/AutomaticMetricQueries/query-generators/default.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    unit,
+  });
+}
+
+export function createDefaultMetricQueryDefs(metricParts: string[]) {
+  // Get the last part of the metric name
+  const suffix = metricParts.at(-1);
+
+  // If the suffix is null or is in the set of unsupported suffixes, throw an error because the metric should be delegated to a different generator (summary or histogram)
+  if (suffix == null || UNSUPPORTED_SUFFIXES.has(suffix)) {
```
Comment (itsmylife) [inline public/app/features/trails/AutomaticMetricQueries/query-generators/default.ts:62]:
```suggestion
  if (!suffix || UNSUPPORTED_SUFFIXES.has(suffix)) {
```

Comment (itsmylife) [reply public/app/features/trails/AutomaticMetricQueries/query-generators/default.ts:62]: I see that double quote `==` check was there before. @darrenjaneczek was this intentional or just forgotten?

Comment (darrenjaneczek) [reply public/app/features/trails/AutomaticMetricQueries/query-generators/default.ts:62]: I often use `== null` to check for `undefined` as well.
I think `!suffix` should be fine.

The only difference, I think, is that now, `""` (empty string) will be considered as well. 
Functionally, this function will work the same.

File: public/app/features/trails/AutomaticMetricQueries/query-generators/index.ts [hunk 2]
```diff
@@ -1,13 +1,20 @@
...
+import { AutoQueryInfo } from '../types';
+
+import { createDefaultMetricQueryDefs } from './default';
+import { createHistogramMetricQueryDefs } from './histogram';
+import { createSummaryMetricQueryDefs } from './summary';
+
+// TODO: when we have a known unit parameter, use that rather than having the generator functions infer from suffix
+export type MetricQueriesGenerator = (metricParts: string[]) => AutoQueryInfo;
+
+export function getQueryGeneratorFor(suffix?: string): MetricQueriesGenerator {
```
[the remark this answers is not in the document]
Comment (yangkb09) [reply public/app/features/trails/AutomaticMetricQueries/query-generators/index.ts:10]: TIL about barrel files! yes absolutely, will rename

**Review by bohandley — APPROVED** [review]
LGTM!

**Review by darrenjaneczek — APPROVED** [review]
:+1:

**Review by itsmylife — APPROVED** [review]

**Pull request merged**